File: vue/src/store/ba04.ts
import { defineStore } from 'pinia'

import {deleteBa04, getBa04List, getBa04ReportVersion, Page} from "../api";
import {ba02Pk} from "./ba02";

export interface ba04Type {
    ba04PK:ba02Pk
    sortOrder: Number
    subOrder: Number
    attr: String
    codeDesc: String
    category: String
    orgCode: String
    addUser: String
    addDate: Date
    addIp: String
    updateUser: String
    updateDate: Date
    updateIp: String
    bigo: String
}
export const Ba04Store = defineStore("Ba04Store",
    {
        state: () => ({
            ba04List:[] as ba04Type[],
            reportVersion:[] as String[]
        }),
        getters: {


        },
        actions: {
            getBa04List(params : Page) {
                return getBa04List(params)
            },
            delete(params : ba04Type[]) {
                return deleteBa04(params)
            },
            getReportVersion()
            {
                const result=getBa04ReportVersion()
                result.then((response) => {
                    this.reportVersion=response
                }).catch((error) => {
                    console.log(error)
                })
            }


        }
    })

File: src/main/java/com/dipol/pdfconvert/data/entity/Ba04.java
package com.dipol.pdfconvert.data.entity;

import lombok.Data;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Data
public class Ba04 {
    @EmbeddedId
    private Ba04PK ba04PK;
    @Basic
    @Column(name = "sort_order")
    private Integer sortOrder;
    @Basic
    @Column(name = "sub_order")
    private Integer subOrder;
    @Basic
    @Column(name = "attr")
    private String attr;
    @Basic
    @Column(name = "code_desc")
    private String codeDesc;
    @Basic
    @Column(name = "category")
    private String category;
    @Basic
    @Column(name = "org_code")
    private String orgCode;
    @Basic
    @Column(name = "add_user")
    private String addUser;
    @Basic
    @Column(name = "add_date")
    private Timestamp addDate;
    @Basic
    @Column(name = "add_ip")
    private String addIp;
    @Basic
    @Column(name = "update_user")
    private String updateUser;
    @Basic
    @Column(name = "update_date")
    private Timestamp updateDate;
    @Basic
    @Column(name = "update_ip")
    private String updateIp;
    @Basic
    @Column(name = "bigo")
    private String bigo;


}
